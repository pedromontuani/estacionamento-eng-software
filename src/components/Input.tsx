import React from 'react';
import {StyleSheet, View} from 'react-native';
import InputMask, {MaskInputProps} from 'react-native-mask-input';

interface IInputProps extends Omit<MaskInputProps, 'value'> {
  value?: string;
}

const Input: React.FC<IInputProps> = ({
  value = '',
  onChangeText,
  placeholder,
  mask,
  ...props
}) => {
  return (
    <View style={styles.container}>
      <InputMask
        value={value}
        onChangeText={onChangeText}
        placeholder={placeholder}
        mask={mask}
        {...props}
      />
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    height: 44,
    width: '100%',
    borderRadius: 22,
    backgroundColor: 'white',
    elevation: 2,
    alignItems: 'stretch',
    justifyContent: 'center',
    paddingHorizontal: 8,
  },
});

export default Input;
