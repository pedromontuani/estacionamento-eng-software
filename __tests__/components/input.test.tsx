import React from 'react';
import {rendererCreate, customRender} from '../utils';
import {fireEvent} from '@testing-library/react-native';

import Input from '../../src/components/Input';

describe('Testing Input component', () => {
  test('should render correctly', async () => {
    const tree = rendererCreate(
      <Input value="" onChangeText={() => {}} />,
    );
    expect(tree).toMatchSnapshot();
  });

  test('should read an input', async () => {
    let value = '';
    const onChangeText = jest.fn().mockImplementation(text => {
      value = text;
    });
    const {getByTestId} = customRender(
      <Input testID="input-test" value={value} onChangeText={onChangeText} />,
    );

    const input = await getByTestId('input-test');

    const MOCK_TEXT = 'a';
    await fireEvent.changeText(input, MOCK_TEXT);

    expect(onChangeText).toHaveBeenCalled();
    expect(value).toBe(MOCK_TEXT);
  });
});
