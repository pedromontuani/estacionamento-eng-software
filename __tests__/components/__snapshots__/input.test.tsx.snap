// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Input component should render correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "white",
      "borderRadius": 22,
      "elevation": 2,
      "height": 44,
      "justifyContent": "center",
      "paddingHorizontal": 8,
      "width": "100%",
    }
  }
>
  <TextInput
    onChangeText={[Function]}
    value=""
  />
</View>
`;
